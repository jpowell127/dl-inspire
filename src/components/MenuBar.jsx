import React from "react";
import PropTypes from "prop-types";
import Search from "./Search";
import { Menu, Icon, Header } from "semantic-ui-react";

function MenuBar({ handleSearchTermChange, getUsers }) {
  return (
    <Menu inverted size="small">
      <Menu.Item>
        <Header as="h1" color="pink">
          <Icon name="behance" />
          <Header.Content>Search</Header.Content>
        </Header>
      </Menu.Item>
      <Menu.Menu position="right">
        <Menu.Item>
          <Search
            handleSearchTermChange={handleSearchTermChange}
            getUsers={getUsers}
          />
        </Menu.Item>
      </Menu.Menu>
    </Menu>
  );
}

Search.MenuBar = {
  getUsers: PropTypes.func.isRequired,
  handleSearchTermChange: PropTypes.func.isRequired
};

export default MenuBar;
