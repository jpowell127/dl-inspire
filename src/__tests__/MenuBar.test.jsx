import React from "react";
import MenuBar from "../components/MenuBar";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("<MenuBar />", () => {
  it("renders correctly", () => {
    const props = {
      getUsers: jest.fn(),
      handleSearchTermChange: jest.fn()
    };
    const component = shallow(<MenuBar {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
