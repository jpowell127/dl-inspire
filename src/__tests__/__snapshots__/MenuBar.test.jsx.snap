// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuBar /> renders correctly 1`] = `
<Menu
  inverted={true}
  size="small"
>
  <MenuItem>
    <Header
      as="h1"
      color="pink"
    >
      <Icon
        as="i"
        name="behance"
      />
      <HeaderContent>
        Search
      </HeaderContent>
    </Header>
  </MenuItem>
  <MenuMenu
    position="right"
  >
    <MenuItem>
      <Search
        getUsers={[MockFunction]}
        handleSearchTermChange={[MockFunction]}
      />
    </MenuItem>
  </MenuMenu>
</Menu>
`;
